pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Requests == nil {
```

Comment:
I think this should have an && limits != nil otherwise it will error if I set limits less than the request you are defaulting errors will occur.  

Also if your goal is to have a Burstable qos and not be prone to eviction, you should check for CPU being set and memory set.  Right no defaulting will not happen if I just set one of them, but maybe that was your goal?

My one other concern is what if I want to run deployer pods with BestEffort qos?  Is there no way to do that or would I have to explicitly set request values to 0?